vim-engine/src/main/kotlin/com/maddyhome/idea/vim/vimscript/model/commands/SourceCommand.kt

```diff
@@ -48,13 +48,23 @@ data class SourceCommand(val range: Range, val modifier: CommandModifier, val ar
     return ExecutionResult.Success
   }
 
-  private fun expandUser(path: String): String {
-    if (path.startsWith("~")) {
+  private fun String.vimExpanded(): String {
+    var expanded = this
+    if (startsWith("~")) {
       val home = System.getProperty("user.home")
       if (home != null) {
-        return home + path.substring(1)
+        expanded = home + substring(1)
       }
     }
-    return path
+
+    val envRe = Regex("\\$[A-Za-z0-9_]+")
+    val env = System.getenv()
```

Comment:
As I see from the help you referencing, this expansion may be also useful in other places. Can I ask you to move it to the injector, to one of the classes?
Also, since this functionality is not trivial, it'll require some tests.